src/classes/vocabulary/jsonld/class-post-jsonld.php

```diff
@@ -115,7 +154,75 @@ public function wl_after_get_jsonld( $jsonld ) {
 		}
 
 		return $jsonld;
-
 	}
 
+	/**
+	 * Set events request.
+	 *
+	 * @param $jsonld_arr array The final jsonld before outputting to page.
+	 * @param $post_id int The post id for which the jsonld is generated.
+	 * @param $context int A context for the JSON-LD generation, valid values in Jsonld_Context_Enum
+	 *
+	 * @return array
+	 */
+	public function set_events_request( $jsonld_arr, $post_id, $context ) {
+		// If context is not PAGE or the array is empty, return early.
+		if ( Jsonld_Context_Enum::PAGE !== $context || empty( $jsonld_arr[0] ) ) {
+			return $jsonld_arr;
+		}
+
+		// Flag to indicate if we should make an API request.
+		$change_status = false;
+
+		// Get data from the array.
+		$data = $jsonld_arr[0];
+
+		// Fetch the initial 'about' and 'mentions' counts from post meta.
+		$counts = array(
+			'about'    => get_post_meta( $post_id, 'wl_about_count', true ) ? get_post_meta( $post_id, 'wl_about_count', true ) : 0,
+			'mentions' => get_post_meta( $post_id, 'wl_mentions_count', true ) ? get_post_meta( $post_id, 'wl_mentions_count', true ) : 0,
+		);
+
+		// Iterate over the counts array.
+		foreach ( $counts as $key => $count ) {
+			// Check if data has 'about' or 'mentions' and the count is different from the existing meta value.
+			if ( ! empty( $data[ $key ] ) ) {
```

Comment:
I think we're missing a case:

1. I have x mentions
2. Event has been sent
3. x mentions recorded in post meta
4. Now mentions is not present anymore (i.e. `empty( $data[ $key ] ) === true`)
5. You should send a count of mentions = 0, the current code won't process it

Same happens later for terms